core/domain/learner_progress_services_test.py

```diff
@@ -481,6 +485,43 @@ def test_get_all_completed_exp_ids(self):
         self.assertEqual(learner_progress_services.get_all_completed_exp_ids(
             self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
+    def test_get_filtered_completed_exp_summaries(self):
+        # Add explorations to the completed list.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_0)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_1)
+
+        # Mark EXP_ID_2 as completed and create summary list to test output.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_2)
+        exploration_summaries = (
+            exp_services.get_exploration_summaries_matching_ids(
+                learner_progress_services.get_all_completed_exp_ids(
+                    self.user_id)))
+
+        # Unpublish EXP_ID_2 to change status to ACTIVITY_STATUS_PRIVATE.
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_exploration(system_user, self.EXP_ID_2)
+        private_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_2)
+        self.assertEqual(
+            private_exploration.status, constants.ACTIVITY_STATUS_PRIVATE)
+
+        # Call get_activity_progress to get filtered progress.
+        user_activity = learner_progress_services.get_activity_progress(
+            self.user_id)
+        all_filtered_summaries = user_activity[0]
+        completed_exp_summaries = (
+            all_filtered_summaries.completed_exp_summaries)
+
+        # Test that completed exp summaries don't include private exploration.
+        self.assertEqual(
+            completed_exp_summaries[0].id, exploration_summaries[0].id)
```

Comment:
Assert this with string value:
`self.assertEqual(completed_exp_summaries[0].id, <some-string-value>)`

Ditto elsewhere!

Reply:
Done!